src/parser.cc

```diff
@@ -1942,19 +1942,27 @@ void copyChildren(ElementPtr _sdf,
       ElementPtr element(new Element);
       element->SetParent(_sdf);
       element->SetName(elem_name);
-      if (elemXml->GetText() != nullptr)
-      {
-        element->AddValue("string", elemXml->GetText(), "1");
-      }
-
+      std::optional<std::string> typeName = std::nullopt;
       for (const tinyxml2::XMLAttribute *attribute = elemXml->FirstAttribute();
            attribute; attribute = attribute->Next())
       {
-        element->AddAttribute(attribute->Name(), "string", "", 1, "");
-        element->GetAttribute(attribute->Name())->SetFromString(
+        const std::string attributeName(attribute->Name());
+        if (attributeName == "type")
```

Comment:
what if a plugin uses the `type` attribute?

we are observing some ign-gazebo test failures of the `triggered_publisher` test, and that plugin uses attributes with name `type`:

* https://github.com/ignitionrobotics/ign-gazebo/pull/1264#issuecomment-1002871765
* https://github.com/ignitionrobotics/ign-gazebo/blob/ign-gazebo6/test/worlds/triggered_publisher.sdf#L5

Reply:
:smile: I had just started writing a similar response here:

---

This broke plugins downstream on `ign-gazebo`. Loading a plugin like [this](https://github.com/ignitionrobotics/ign-gazebo/blob/fe335f9cd47e7f7b3ff35159babe28d9fd55670d/test/worlds/triggered_publisher.sdf#L9-L14):

```
    <plugin filename="ignition-gazebo-triggered-publisher-system" name="ignition::gazebo::systems::TriggeredPublisher">
      <input type="ignition.msgs.Empty" topic="/in_1"/>
      <output type="ignition.msgs.Boolean" topic="/out_1">
          data: true
      </output>
    </plugin>
```

Results in an assertion when the SDF is loaded:

```
173: Error [Param.cc:768] Unknown parameter type[ignition.msgs.Boolean]
173: Exception [Param.cc:81] SDF ASSERTION                     
173: Invalid parameter
173: In function       : Param
173: Assert expression : this->dataPtr->ValueFromStringImpl( this->dataPtr->typeName, _default, this->dataPtr->defaultValue)
```

I'm testing a temporary fix

Reply:
This is what I'm trying: https://github.com/ignitionrobotics/sdformat/pull/809